guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -51,78 +50,78 @@
 import org.eclipse.aether.util.artifact.SubArtifact;
 import org.guvnor.common.services.project.model.GAV;
 import org.guvnor.m2repo.backend.server.repositories.ArtifactRepository;
-import org.guvnor.m2repo.backend.server.repositories.DistributionManagementArtifactRepository;
-import org.guvnor.m2repo.backend.server.repositories.FileSystemArtifactRepository;
-import org.guvnor.m2repo.backend.server.repositories.LocalArtifactRepository;
-import org.guvnor.m2repo.preferences.ArtifactRepositoryPreference;
+import org.guvnor.m2repo.backend.server.repositories.ArtifactRepositoryFactory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.uberfire.apache.commons.io.FilenameUtils;
 
-import static org.guvnor.m2repo.utils.FileNameUtilities.*;
+import static org.guvnor.m2repo.utils.FileNameUtilities.isDeployedPom;
+import static org.guvnor.m2repo.utils.FileNameUtilities.isJar;
+import static org.guvnor.m2repo.utils.FileNameUtilities.isKJar;
 
 @ApplicationScoped
 public class GuvnorM2Repository {
 
     private static final Logger log = LoggerFactory.getLogger(GuvnorM2Repository.class);
 
-    public static String M2_REPO_DIR;
-
     private static final int BUFFER_SIZE = 1024;
 
     private final List<ArtifactRepository> repositories = new ArrayList<>();
     private final List<ArtifactRepository> pomRepositories = new ArrayList<>();
-
-    private ArtifactRepositoryPreference preferences;
+    private ArtifactRepositoryFactory artifactRepositoryFactory;
 
     public GuvnorM2Repository() {
     }
 
     @Inject
-    public GuvnorM2Repository(ArtifactRepositoryPreference preferences) {
-        this.preferences = preferences;
+    public GuvnorM2Repository(ArtifactRepositoryFactory factory) {
+        this.artifactRepositoryFactory = factory;
     }
 
     @PostConstruct
     public void init() {
-        preferences.load();
         setM2Repos();
     }
 
     private void setM2Repos() {
-        final String M2_REPO_ROOT = FilenameUtils.separatorsToSystem(preferences.getDefaultM2RepoDir());
 
-        final String meReposDir = System.getProperty("org.guvnor.m2repo.dir");
+        this.repositories.addAll(this.artifactRepositoryFactory.getRepositories());
+        this.pomRepositories.addAll(this.artifactRepositoryFactory.getPomRepositories());
```

Comment:
@manstis they are used into deployArtifact and deployPom because for instance pom is not deployed through distributionManagement, so I had to do that difference.